platform/darwin/test/MGLStyleTests.mm

```diff
@@ -427,6 +424,36 @@ - (void)testLayersOrder {
 
 #pragma mark Localization tests
 
+- (void)testLocalization {
+    MGLSymbolStyleLayer *countryLabel = (MGLSymbolStyleLayer *)[self.style layerWithIdentifier:@"country-label"];
+    {
+        NSLocale *locale = [NSLocale localeWithLocaleIdentifier:@"de_DE"];
+        [self.style localizeLabelsIntoLocale:locale];
+
+        NSArray * keypathArray = @[ [NSExpression expressionForKeyPath:@"name_de"],
+                                    [NSExpression expressionForKeyPath:@"name"]];
+        NSExpression *coalesceExpression = [NSExpression expressionWithFormat:@"mgl_coalesce:(%@)", @[ [NSExpression expressionWithFormat:@"mgl_coalesce:(%@)", keypathArray],
+                                                                                                       [NSExpression expressionWithFormat:@"mgl_coalesce:(%@)", keypathArray] ]];
+        MGLAttributedExpression *attributedExpression = [MGLAttributedExpression attributedExpression:coalesceExpression attributes:@{}];
+        NSExpression *localizedExpression = [NSExpression mgl_expressionForAttributedExpressions:@[ [NSExpression expressionForConstantValue:attributedExpression] ]];
+        XCTAssertEqualObjects(countryLabel.text, localizedExpression);
+        countryLabel.text = [NSExpression expressionWithFormat:@"mgl_coalesce({%K, %K})", @"name_en", @"name"];
```

Comment:
I think it’s more important to test that `-[NSExpression mgl_expressionLocalizedIntoLocale:]` returns the right value than to test the full integration with `-[MGLStyle localizeLabelsIntoLocale:]`, especially considering the difficulty of removing network dependencies.